timesketch/lib/charts/barchart.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Barcharts."""
+
+from __future__ import unicode_literals
+
+import altair as alt
+
+from timesketch.lib.charts import manager
+from timesketch.lib.charts import interface
+
+
+class BarChart(interface.BaseChart):
+    """Barchart object."""
+
+    NAME = 'barchart'
+
+    def __init__(self, data):
+        """Initialize the chart object.
+
+        Args:
+            data: Aggregation result from an aggregator object as a dict.
+        """
+        super(BarChart, self).__init__(data)
+
+    def generate(self):
+        """Generate the chart.
+
+        Returns:
+            Instance of altair.Chart
+        """
+        chart = alt.Chart(self.values).mark_bar()
+        chart.encoding = alt.Encoding.from_dict(self.encoding)
```

Comment:
are we always certain there is some encoding set?

Reply:
Fixed.

Reply:
is it? ATM it looks like encoding can be empty.... or None...